testng-eclipse-plugin/src/main/org/testng/eclipse/launch/TestNGLaunchConfigurationDelegate.java

```diff
@@ -329,21 +329,25 @@ protected VMRunnerConfiguration createVMRunner(final ILaunchConfiguration config
   public String[] getClasspath(ILaunchConfiguration configuration)
       throws CoreException {
     String[] originalClasspath = super.getClasspath(configuration);
+    // add remote testng jar file at the very begining to make sure this is loaded in prior of the user's testng.jar
+    String[] classpathArray = new String[originalClasspath.length + 1];
+    classpathArray[0] = BuildPathSupport.getRemoteTestNGLibPath();

```

Comment:
add remote testng jar file at the very begining to make sure this is loaded in prior of the user's testng.jar